### PR TITLE
tests: qa/tasks/ceph.py: fail if any osd role doesn't get a device

### PR DESCRIPTION
**Pull request by smithfarm**
Fixes: https://tracker.ceph.com/issues/42357

### CODE REVIEW

File: qa/tasks/ceph.py [hunk 1]
```diff
@@ -607,44 +607,23 @@ def cluster(ctx, config):
...
-            remote.run(args=['sudo', 'mount', '-t', 'tmpfs', 'tmpfs', '/mnt'])
-            for role in teuthology.cluster_roles_of_type(roles_for_host, 'osd', cluster_name):
-                tmpfs = '/mnt/' + role
-                roles_to_journals[role] = tmpfs
-                remote.run(args=['truncate', '-s', '1500M', tmpfs])
-            log.info('journal map: %s', roles_to_journals)
-
-        log.info('dev map: %s' % (str(roles_to_devs),))
+        fs_val = config.get('fs')
+        assert fs_val, "remote {} has osd roles, but fs option was not set!"
```
Comment (kshtsk) [inline qa/tasks/ceph.py]: missing .format?

Comment (smithfarm) [reply qa/tasks/ceph.py]: good catch - fixed

Comment (kshtsk) [reply qa/tasks/ceph.py]: oh... the assert message should be rewritten, since it is out of the remotes loop

**Comment by jdurgin** [comment]
'ceph osd metadata [osd.x]' shows the disks used by an osd

File: qa/tasks/ceph.py [hunk 2]
```diff
@@ -619,27 +619,7 @@ def cluster(ctx, config):
             roles_to_devs = assign_devs(
                 teuthology.cluster_roles_of_type(roles_for_host, 'osd', cluster_name), devs
             )
-            if len(roles_to_devs) < len(devs):
-                devs = devs[len(roles_to_devs):]
             devs_to_clean[remote] = []
-
-        if config.get('block_journal'):
```
Comment (jdurgin) [inline qa/tasks/ceph.py]: these were mainly for filestore testing a long time ago - fine to drop now imo

File: qa/tasks/ceph.py [hunk 3]
```diff
@@ -607,12 +607,10 @@ def cluster(ctx, config):
 
     devs_to_clean = {}
     remote_to_roles_to_devs = {}
-    remote_to_roles_to_journals = {}
```
Comment (jdurgin) [inline qa/tasks/ceph.py]: :+1: these were already a bit obsolete with our uniform-media machines in sepia. especially once ceph-daemon uses ceph-volume, it makes no sense to keep these around.

File: qa/tasks/ceph.py [hunk 4]
```diff
@@ -611,15 +611,20 @@ def cluster(ctx, config):
...
         roles_to_devs = {}
-        if config.get('fs'):
-            log.info('fs option selected, checking for "scratch" (osd) devs')
-            log.info('found "scratch" (osd) devs: {}'.format(devs))
-            roles_to_devs = assign_devs(
-                teuthology.cluster_roles_of_type(roles_for_host, 'osd', cluster_name), devs
-            )
-            devs_to_clean[remote] = []
+        fs_val = config.get('fs')
+        assert fs_val, \
```
Comment (jdurgin) [inline qa/tasks/ceph.py]: fs doesn't apply to bluestore (even if configs happen to have it, it makes no sense since bluestore uses the raw block device)

**Comment by smithfarm** [comment]
I was thinking about:

* programatically determine the remote's root filesystem device
* as we loop over the remote's osd devices, assert that they aren't the root device

**Comment by gregsfortytwo** [comment]
👍

File: qa/tasks/ceph.py (hunk 4 above)
Comment (tchaikov) [reply qa/tasks/ceph.py]: @smithfarm i concur with Josh that "fs" is not necessary when it comes bluestore. i don't think we need to check it and assert its existence.

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2019-10-25_07:20:39-rados-wip-kefu-testing-2019-10-25-0853-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2019-10-26_00:09:34-rados-wip-kefu-testing-2019-10-25-0853-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2019-10-26_15:51:16-rados-wip-kefu-testing-2019-10-26-1857-distro-basic-mira/

**Comment by smithfarm** [comment]
@gregsfortytwo @tchaikov @jdurgin Please take another look - I removed the offending lines.

**Comment by gregsfortytwo** [comment]
LGTM at a quick skim.

**Comment by tchaikov** [comment]
@gregsfortytwo if the latest revision addresses your request could you lift the "request changes" bar?

**Pull request merged**